packages/signers/src/lit-protocol/signer.ts

```diff
@@ -0,0 +1,217 @@
+import { PKPEthersWallet } from "@lit-protocol/pkp-ethers";
+import { LitAbility, LitPKPResource } from "@lit-protocol/auth-helpers";
+import {
+  type AuthCallbackParams,
+  type AuthSig,
+  type LITEVMChain,
+  type SessionSigsMap,
+} from "@lit-protocol/types";
+import { LitNodeClient } from "@lit-protocol/lit-node-client";
+import { ALL_LIT_CHAINS } from "@lit-protocol/constants";
+import { generateSessionKeyPair } from "@lit-protocol/crypto";
+import { type SignTypedDataParams } from "@alchemy/aa-core";
+import type { Address, Hex, TypedDataDomain } from "viem";
+import {
+  type LitAuthMethod,
+  type LitSessionSigsMap,
+  type LitConfig,
+  type LitAuthenticateProps,
+  type LitSmartAccountAuthenticator,
+  type LitUserMetadata,
+} from "./types.js";
+import { signerTypePrefix } from "../constants.js";
+
+const SIGNER_TYPE: string = `${signerTypePrefix}lit`;
+
+/**
+ * Implementation of `SmartAccountAuthenticator` for lit protocol
+ * This class requies:
+ * `@lit-protocol/lit-node-client@cayenne`
+ * `@lit-protocol/pkp-ethers@cayenne`
+ * `@lit-protocol/crypto@cayenne`
+ * `@lit-protocol/auth-helpers@cayenne`
+ * `@lit-protocol/types@cayenne`
+ */
```

Comment:
is the @cayenne necessary?

Reply:
Yes for now it is as we have not turned on features in our `latest` version tag.
Once we publish latest we can come back and remove the `cayenne` tag